modifier/modifier.go

```diff
@@ -66,3 +67,21 @@ func newExtendedLayerPart(layerPart data.LayerPart, typ ...string) extendedLayer
 func (l extendedLayerPart) Attributes() map[string]interface{} {
 	return l.attributes
 }
+
+// PartsAttribute extracts the given attribute from the layer.
+// It supports only []data.LayerPart as type.
+// If it has the wrong type, a error is returned.
+// If it doesn't exist, (nil, nil) is returned.
+// If it exists, the infill is returned.
+func PartsAttribute(layer data.PartitionedLayer, typ string) ([]data.LayerPart, error) {
```

Comment:
The parameter name should be `type` I guess.

Reply:
nope as type is a reserved keyword it doesn't work